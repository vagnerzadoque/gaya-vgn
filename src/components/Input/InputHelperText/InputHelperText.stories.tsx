import React from 'react'
import { Story, Meta } from '@storybook/react'
import { InputHelperText, InputHelperTextProps } from '.'

export default {
  title: 'Utilities/InputHelperText',
  component: InputHelperText,
  parameters: {
    previewTabs: {
      canvas: { hidden: true }
    }
  }
} as Meta

export const Default: Story<InputHelperTextProps> = (args) => <InputHelperText {...args} />
Default.args = {
  helperText: 'Helper Text'
}
