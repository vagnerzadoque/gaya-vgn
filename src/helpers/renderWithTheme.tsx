import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import themes from '@naturacosmeticos/natds-themes'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component: JSX.Element) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider theme={themes.natura.light}>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
