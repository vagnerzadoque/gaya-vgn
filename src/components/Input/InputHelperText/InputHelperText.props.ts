import { InputProps } from '../Input.props'

export type InputHelperTextProps = Pick<InputProps, 'feedback' | 'helperText' | 'disabled'>
