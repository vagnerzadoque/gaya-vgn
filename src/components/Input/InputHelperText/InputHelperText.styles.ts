/* eslint-disable max-len */
import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { InputHelperTextProps } from './InputHelperText.props'

type InputHelperTextStyleProps = Pick<InputHelperTextProps, 'feedback' | 'disabled'>

export const getFeedbackTextColor = (theme: Theme) => ({ feedback, disabled }: InputHelperTextStyleProps) => {
  const defaultColor = disabled ? theme.color.lowEmphasis : theme.color.mediumEmphasis

  const textColor = {
    error: theme.color.alert,
    success: theme.color.success
  }

  return feedback ? textColor[feedback] : defaultColor
}

const styles = createUseStyles((theme: Theme) => ({
  helperText: {
    alignItems: 'center',
    color: getFeedbackTextColor(theme),
    display: 'flex',
    fontFamily: [
      theme.textField.helperText.primary.fontFamily,
      theme.textField.helperText.fallback.fontFamily
    ],
    fontSize: theme.textField.helperText.fontSize,
    fontWeight: theme.textField.helperText.primary.fontWeight,
    letterSpacing: theme.textField.helperText.letterSpacing,
    lineHeight: theme.textField.helperText.lineHeight,
    marginBottom: 0,
    marginTop: theme.spacing.micro,
    '& > i': {
      marginRight: ({ feedback }: InputHelperTextStyleProps) => feedback && theme.spacing.micro
    }
  }
}))

export default styles
