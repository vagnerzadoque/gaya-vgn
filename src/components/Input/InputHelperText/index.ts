import InputHelperText from './InputHelperText'

export { InputHelperTextProps } from './InputHelperText.props'
export { InputHelperText }
export default InputHelperText
