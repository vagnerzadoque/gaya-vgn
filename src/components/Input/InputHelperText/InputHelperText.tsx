import React from 'react'
import styles from './InputHelperText.styles'
import { Icon } from '../../Icon'
import { InputHelperTextProps } from './InputHelperText.props'
import { Feedback } from '../Input.props'

export const getIconName = (feedback: Feedback) => (feedback === 'success' ? 'outlined-action-check' : 'outlined-action-cancel')
export const getIconColor = (feedback: Feedback) => (feedback === 'success' ? 'success' : 'alert')

const InputHelperText = ({
  disabled,
  feedback,
  helperText
}: InputHelperTextProps): JSX.Element => {
  const classes = styles({ feedback, disabled })

  return (
    <p className={classes.helperText}>
      {feedback && <Icon name={getIconName(feedback)} color={getIconColor(feedback)} size="small" />}
      {helperText}
    </p>
  )
}

export default InputHelperText
