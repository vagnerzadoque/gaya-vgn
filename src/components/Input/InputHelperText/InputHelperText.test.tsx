import React from 'react'
import renderWithTheme from '../../../helpers/renderWithTheme'
import InputHelperText, { getIconColor, getIconName } from './InputHelperText'

describe('InputHelperText component', () => {
  it('should render correctly with helper text', () => {
    const { styles, component } = renderWithTheme(<InputHelperText helperText="some helper text" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with helper text and feedback error', () => {
    const { styles, component } = renderWithTheme(<InputHelperText helperText="some helper text" feedback="error" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with helper text and feedback success', () => {
    const { styles, component } = renderWithTheme(<InputHelperText helperText="some helper text" feedback="success" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})

describe('getIconName', () => {
  it('should return outlined-action-check when receive feedback success', () => {
    expect(getIconName('success')).toBe('outlined-action-check')
  })
  it('should return outlined-action-cancel when receive feedback error', () => {
    expect(getIconName('error')).toBe('outlined-action-cancel')
  })
})

describe('getIconColor', () => {
  it('should return `success` when receive feedback success', () => {
    expect(getIconColor('success')).toBe('success')
  })
  it('should return `alert` when receive feedback error', () => {
    expect(getIconColor('error')).toBe('alert')
  })
})
