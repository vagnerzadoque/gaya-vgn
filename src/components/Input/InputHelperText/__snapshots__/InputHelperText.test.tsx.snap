// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHelperText component should render correctly with helper text 1`] = `
[
  ".helperText-0-2-1 {
  display: flex;
  font-size: 12px;
  margin-top: 4px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 400;
  line-height: 1.25;
  margin-bottom: 0;
  letter-spacing: 0.38px;
}
.helperText-0-2-1 > i {}
.helperText-d0-0-2-2 {
  color: #737373;
}
.helperText-d0-0-2-2 > i {}",
  <div>
    <p
      class="helperText-0-2-1 helperText-d0-0-2-2"
    >
      some helper text
    </p>
  </div>,
]
`;

exports[`InputHelperText component should render correctly with helper text and feedback error 1`] = `
[
  ".helperText-0-2-1 {
  display: flex;
  font-size: 12px;
  margin-top: 4px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 400;
  line-height: 1.25;
  margin-bottom: 0;
  letter-spacing: 0.38px;
}
.helperText-0-2-1 > i {}
.helperText-d0-0-2-2 {
  color: #de3529;
}
.helperText-d0-0-2-2 > i {
  margin-right: 4px;
}
.icon-0-2-3 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-4 {
  color: #de3529;
  font-size: 16px;
}",
  <div>
    <p
      class="helperText-0-2-1 helperText-d0-0-2-2"
    >
      <i
        aria-hidden="true"
        class=" icon-0-2-3 icon-d0-0-2-4 natds-icons natds-icons-outlined-action-cancel Icon-gaya"
        data-testid="icon-outlined-action-cancel"
        role="img"
      />
      some helper text
    </p>
  </div>,
]
`;

exports[`InputHelperText component should render correctly with helper text and feedback success 1`] = `
[
  ".helperText-0-2-1 {
  display: flex;
  font-size: 12px;
  margin-top: 4px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 400;
  line-height: 1.25;
  margin-bottom: 0;
  letter-spacing: 0.38px;
}
.helperText-0-2-1 > i {}
.helperText-d0-0-2-2 {
  color: #2f833e;
}
.helperText-d0-0-2-2 > i {
  margin-right: 4px;
}
.icon-0-2-3 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-4 {
  color: #2f833e;
  font-size: 16px;
}",
  <div>
    <p
      class="helperText-0-2-1 helperText-d0-0-2-2"
    >
      <i
        aria-hidden="true"
        class=" icon-0-2-3 icon-d0-0-2-4 natds-icons natds-icons-outlined-action-check Icon-gaya"
        data-testid="icon-outlined-action-check"
        role="img"
      />
      some helper text
    </p>
  </div>,
]
`;
